org.eclipse.jdt.ls.core/gradle/android/init.gradle

```diff
@@ -0,0 +1,430 @@
+                            ((DefaultProjectDependency) dependency).setTargetConfiguration("default")
+                        }
+                    }
+                }
+                configuration.setCanBeResolved(true)
+                eclipseModel.classpath.plusConfigurations.add(configuration)
+            }
+        }
+    }
+
+    private static Object getAndroidSourceSetByName(Project project, String name) {
+        Object android = project.property(ANDROID_PROPERTY)
+        if (android.properties.containsKey("sourceSets")) {
+            Object sourceSets = android.properties.get("sourceSets")
+            if (sourceSets instanceof DefaultNamedDomainObjectCollection) {
+                DefaultNamedDomainObjectCollection sourceCollection = ((DefaultNamedDomainObjectCollection) sourceSets)
+                if (sourceCollection.names.contains(name)) {
+                    return sourceCollection.getByName(name)
+                }
+            }
+        }
+        return null
+    }
+
+    private static List<Object> getAndroidDebuggableVariants(Project project) {
+        Object android = project.property(ANDROID_PROPERTY)
+        Object variants = null
+        if (project.plugins.hasPlugin("com.android.application")) {
+            // For android application, variants come from applicationVariants property
+            if (android.properties.containsKey("applicationVariants")) {
+                variants = android.properties.get("applicationVariants")
+            }
+        } else if (project.plugins.hasPlugin("com.android.library")) {
+            // For android library, variants come from libraryVariants property
+            if (android.properties.containsKey("libraryVariants")) {
+                variants = android.properties.get("libraryVariants")
+            }
+        }
+        if (variants instanceof DefaultDomainObjectCollection) {
+            return ((DefaultDomainObjectCollection) variants).stream().filter(variant -> {
+                try {
+                    Object debuggable = variant.properties.get("buildType").properties.get("debuggable")
+                    return debuggable instanceof Boolean && ((Boolean) debuggable)
+                } catch (NullPointerException ignored) {
+                    return false
+                }
+            }).collect(Collectors.toList())
+        }
+        return Collections.emptyList()
+    }
+
+    private static void addFolderToSourceFolder(Classpath classpath, Project project, File folder, String outputPath) {
+        if (!folder.exists()) {
+            if (!folder.mkdirs()) {
+                return
+            }
+        }
+        if (!folder.isDirectory()) {
+            return
+        }
+        SourceFolder sourceFolder = new SourceFolder(project.getProjectDir().toURI().relativize(folder.toURI()).toString(), outputPath)
+        sourceFolder.entryAttributes.put(OPTIONAL_ATTRIBUTE, "true")
```

Comment:
Is the optional attribute aimed for Gradle versions under 5.4?

Reply:
Yes. For Gradle versions greater than 5.4 we will generate these folders automatically, let me add check later.